astrocut/cube_cut_from_footprint.py

```diff
@@ -0,0 +1,310 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+
+"""This module creates cutouts from data cubes found in the cloud."""
+
+import json
+import re
+import os
+from typing import List, Literal, Union
+import warnings
+from concurrent.futures import ThreadPoolExecutor, as_completed
+
+from astropy.coordinates import SkyCoord
+from astropy.table import Table, Column
+import astropy.units as u
+from astropy.utils.exceptions import AstropyWarning
+import numpy as np
+import fsspec
+from spherical_geometry.polygon import SphericalPolygon
+from spherical_geometry.vector import radec_to_vector
+
+from astrocut.exceptions import InvalidQueryError
+from astrocut.cube_cut import CutoutFactory
+
+from .utils.utils import parse_size_input
+
+TESS_ARCSEC_PER_PX = 21
+
+
+def _s_region_to_polygon(s_region: Column):
+    """
+    Takes in a s_region string of type POLYGON or CIRCLE and returns it as
+    a spherical_region Polygon.
+
+    Example inputs:
+    'POLYGON 229.80771900 -75.17048500 241.67788000 -63.95992300 269.94872000 -64.39276400 277.87862300 -75.57754400'
+    'CIRCLE ICRS 244.38324081 -75.86611807 0.625'
+    """
+
+    def ind_sregion_to_polygon(s_reg):
+        sr_list = s_reg.strip().split()
+        reg_type = sr_list[0].upper()
+
+        if reg_type == "POLYGON":
+            ras = np.array(sr_list[1::2], dtype=float)
+            ras[ras < 0] = ras[ras < 0] + 360
+            decs = np.array(sr_list[2::2], dtype=float)
+            return SphericalPolygon.from_radec(ras, decs)
+        elif reg_type == "CIRCLE":
+            ra, dec, rad = np.array(sr_list[-3:], dtype=float)
+            return SphericalPolygon.from_cone(ra, dec, rad)
+        else:
+            raise ValueError("Unsupported S_Region type.")
+
+    return np.vectorize(ind_sregion_to_polygon)(s_region)
+
+
+def _get_s3_ffis(s3_uri, as_table: bool = False, load_polys: bool = False):
+    """
+    Fetch the S3 footprint file containing a dict of all FFIs and a polygon column
+    that holds the s_regions as polygon points and vectors.
+
+    Optional Parameters:
+        as_table: Return the footprint file as an Astropy Table
+        load_polys: Convert the s_region column to an array of SphericalPolygon objects
+    """
+    # Open footprint file with fsspec
+    # Use caching to help performance, but check that remote UID matches the local
+    # Expiry time is 1 week by default
+    with fsspec.open('filecache::' + s3_uri, s3={'anon': True}, 
+                     filecache={'cache_storage': 's3_cache', 'check_files': True}) as f:
+        ffis = json.loads(f.read())
```

Comment:
```suggestion
        ffis = json.load(f)
```

Reply:
I get a `TypeError` from this change because `json.loads()` won't accept a `BufferedReader` object.